intellij/src/nl/tudelft/watchdog/intellij/logic/ui/listeners/staticanalysis/CheckStyleStartup.java

```diff
@@ -0,0 +1,90 @@
+
+/**
+ * Component that is lazily initiated, only if the CheckStyle-IDEA plugin is actually available in the editor.
+ * It loads the default configured messages from CheckStyle (in their `messages.properties` files) and
+ * traverses the currently activated configuration for any custom messages.
+ */
+public class CheckStyleStartup implements ProjectComponent {
+
+    private final Project project;
+
+    public CheckStyleStartup(@NotNull final Project project) {
+        this.project = project;
+    }
+
+    @Override
+    public void initComponent() {
+        try {
+            CheckstyleProjectService service = ServiceManager.getService(this.project, CheckstyleProjectService.class);
+
+            // This loader can load all checks as defined in CheckStyle
+            final ClassLoader checkStylePackageLoaderClassLoader = service.getCheckstyleInstance().getClass().getClassLoader();
+
+            // This loader is used to obtain the actual resources
+            final ClassLoader checkStylePluginClassLoader = getPluginCreatedClassLoaderFromService(service);
+
+            addCheckStyleMessagesToBundle(checkStylePackageLoaderClassLoader, checkStylePluginClassLoader);
+            addMessagesForActiveConfiguration(service);
+
+            StaticAnalysisMessageClassifier.CHECKSTYLE_BUNDLE.sortList();
+        } catch (Exception e) {
+            WatchDogLogger.getInstance().logSevere("Could not initialize the CheckStyle plugin. This is likely an issue with an outdated version of the CheckStyle plugin: " + e);
+        }
+    }
+
+    // TODO(timvdlippe): update the messages when the configuration changes
+    private void addMessagesForActiveConfiguration(CheckstyleProjectService service) {
+        final CheckStylePlugin checkStylePlugin = project.getComponent(CheckStylePlugin.class);
+        final ConfigurationLocation activeConfigLocation = checkStylePlugin.configurationManager().getCurrent().getActiveLocation();
+
+        if (activeConfigLocation == null) {
+            return;
+        }
+
+        service.getCheckstyleInstance().peruseConfiguration(
+                service.getCheckstyleInstance()
+                        .loadConfiguration(activeConfigLocation, true, new HashMap<>()),
+                module -> {
+                    String moduleKey = activeConfigLocation.getDescription() + "." + module.getName() + ".";
+
+                    for (Map.Entry<String, String> message: module.getMessages().entrySet()) {
+                        addMessageToCheckstyleBundle(moduleKey + message.getKey(), message.getValue());
+                    }
+                }
+        );
+    }
+
+    // A bunch of reflection magic, as the plugin does not expose its internal classLoader
+    // TODO(timvdlippe): Update this code to use the type-safe methods once a PR to the plugin is merged that implements the method.
```

Comment:
also here

Reply:
You could potentially leave a pointer to the PR you implemented, so people understand where this comes from if you do not get around to doing it

Reply:
I will get to this after this PR, so it will be removed in no time